src/csync/csync_exclude.cpp

```diff
@@ -291,8 +291,14 @@ void ExcludedFiles::setClientVersion(ExcludedFiles::Version version)
 bool ExcludedFiles::loadExcludeFile(const QString &basePath, const QString & file)
 {
     QFile f(file);
-    if (!f.open(QIODevice::ReadOnly))
+    if (!f.open(QIODevice::ReadOnly)) {
+        if (!f.exists()) {
+            _excludeFiles.remove(basePath);
```

Comment:
Since `loadExcludeFile()` itself isn't doing anything with `_excludeFiles and since `_excludeFiles` is really only used inside `reloadsExcludeFiles()`, wouldn't it be better to have that fix in `reloadExcludeFiles()`?

By the way I noticed the following, in `reloadExcludeFiles()` the `success` variable can only be false when the last file failed at loading, but none of the other ones, that sounds wrong to me. The `=` on line 334 should likely be replaced by `&=`.

Finally, I'm wondering (totally unsure) if we want to just skip the file when it doesn't exist instead of removing it from the _excludeFiles list. In the case of a folder subscription for instance, could it somehow come back if you select the folder again?

Reply:
Also, probably a good idea to extend the test suite for that class to also cover that case.